### DISCUSSION TITLE
New Documentation Site

### DISCUSSION THREAD

**Discussion by babakfp** [colinhacks/zod#3019] [Show and tell]
Hi

I'm working on this new documentation site for Zod:

https://zod-eight.vercel.app

![Screenshot 2023-12-17 072733-min](https://github.com/colinhacks/zod/assets/44144724/5a9adfdf-ad6a-491f-b4cf-7af43a21630e)

The problems I have with the current site are:

- Bad SEO (sometimes can't find things in docs when searching on Google)
- Slow
- Bad UI

I'm using [VitePress](https://vitepress.dev) for documentation generation, which is fantastic.

---

### Content

I moved all of the `.md` files to the new site. So, if any changes need to be made to the files like `README.md`, `MIGRATION.md`, `CONTRIBUTING.md` (etc), they need to be made in the new sites Markdown files.

### Sponsors

About sponsors. I don't know how these changes are going to affect the impressions promised in [this page](https://github.com/sponsors/colinhacks). Are these changes going to impact the number of sponsors or the promised impressions number in a bad way? I don't know.

On the new site, there are three places the sponsors are shown, the home page, the sponsors page, also the main guide page (the page that contains content of README.md).

I think it would be nice to show all sponsors on the site, but it seems like GitHub doesn't provide an API for it (maybe I couldn't find it). So, does anyone know how to do this?

### i18n

I don't know `zh`, but it seemed like it needs some rewrites. Maybe it's a good idea to rewrite the original docs and after that update the `zh` docs too.

### Issues

If the user changes the language on a page that is not yet translated, then a 404 error is going to be shown, so there is no fallback to the default language. There is a workaround but it's not a nice one.

https://github.com/vuejs/vitepress/pull/3312

---

- [ ] ~I would also link to build a playground (https://github.com/colinhacks/zod/issues/2710).~
I tried implementing this but things didn't work...

- [ ] Which Discord link to pick? https://discord.gg/KaSRdyX2vc or https://discord.gg/RcG33DQJdf.
One of the links is used here:
![image](https://github.com/colinhacks/zod/assets/44144724/3e20993e-36d4-47d3-8b90-8a98c339c30d)
And the other one is used here:
![image](https://github.com/colinhacks/zod/assets/44144724/28cceb9d-92fa-4ae1-b73d-64f38ea91fb3)
Should I keep them as is in the new docs, or only use one? they both link to the same discord server.

- [ ] Replace `|` with `-` if possible. (SEO).

- [ ] gtag.
I may be wrong, but this code looks wrong 🤔:
```
<script
    async
    src="https://www.googletagmanager.com/gtag/js?id=G-FG8DDV0GBR"
></script>
<script>
    window.dataLayer = window.dataLayer || [];
    function gtag() {
    dataLayer.push(arguments);
    }
    gtag("js", new Date());
    gtag("config", "G-FG8DDV0GBR");
</script>
```

**Comment by JacobWeisenburger** [comment]
This seems more like a discussion, is it ok if I move it there?

**Comment by babakfp** [comment]
> This seems more like a discussion, is it ok if I move it there?

Hi. Sure, feel free to do so.

**Comment by babakfp** [comment]
@colinhacks

Hey. Can you please take a look into this? Thanks.

Things are up to date with the latest commit on this repo.
